src/Socket/Socket.cpp

```diff
@@ -0,0 +1,71 @@
+#include "Socket.hpp"
+
+Socket::Socket(int domain, unsigned short port, int type, int protocol, u_long interface)
+{
+	_address.sin_family = domain;
+	_address.sin_port = htons(port);
+	_address.sin_addr.s_addr = htonl(interface);
+
+	create_socket(domain, type, protocol);
+	set_socket_non_blocking();
+	binding_socket();
+	start_listening();
+}
+
+void
+Socket::create_socket(int domain, int type, int protocol)
+{
+	_sock_id = socket(domain, type, protocol);
+	test_socket(_sock_id, "create_socket() Fail!");
+}
+
+void
+Socket::binding_socket()
+{
+	_connection = bind(_sock_id, reinterpret_cast<struct sockaddr *>(&_address), sizeof(_address));
+	test_socket(_connection, "binding_socket() Fail!");
+}
+
+void
+Socket::start_listening()
+{
+	int res;
+	res = listen(_sock_id, 10);
+	test_socket(res, "start_listening() Fail!");
+}
+
+void
+Socket::test_socket(int item_to_test, const char *msg)
+{
+	if (item_to_test < 0)
+	{
+		std::cerr << msg << std::endl;
+		close(_sock_id);
+		throw std::runtime_error("Exception: Socket set up failed.");
```

Comment:
What do you think about print the message once in the exception ?
```suggestion
		close(_sock_id);
		throw std::runtime_error(msg);
```

Reply:
yes of course